lib/translator/common.h

```diff
@@ -44,6 +46,13 @@ void DeallocPages(void* pages_ptr, uint16_t count);
 void SetAllocPageCallbacks(void* (*alloc_callback)(uint16_t),
                            void (*dealloc_callback)(void*, uint16_t));
 
+struct allocation {
+  void* data_addr;      // start of data buffer
+  uint16_t data_size;   // data page size
+  void* mdata_addr;     // start of metadata buffer
+  uint16_t mdata_size;  // metadata page size
```

Comment:
I'm a bit confused by the notion of "page size" here. This may be because I'm lacking some context, but if so some more comments might be justified:

Aren't pages more or less fixed-size? Are these the size of one page, a quantity of pages, or the size of a single allocation that may span multiple pages? If the last, why even describe them in terms of pages, since that's just standard memory allocation?

Reply:
Poor word choice on my part. These are intended to capture the number of pages that were allocated for the NVMe buffer. Pages were chosen over standard memory allocation because NVMe responses are almost always multiples of pages. I'll update with clearer phrasing!

Reply:
Thanks, that's clearer!